CMakeLists.txt

```diff
@@ -59,8 +67,8 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "(Clang|GNU|Intel)")
             "Flags used by the compiler during debug builds.")
 
     # Add a build type that keeps runtime checks enabled
-    set(CMAKE_CXX_FLAGS_RELWITHASSERT "-O3" CACHE INTERNAL
-            "Flags used by the compiler during release builds containing runtime checks.")
+    set(CMAKE_CXX_FLAGS_RELWITHASSERT "-O3 -D_GLIBCXX_DEBUG=1" CACHE STRING
```

Comment:
:shrug: It would not be unexpected IMHO for "with assertions". A lighter variant is `-D_GLIBCXX_ASSERTIONS=1` (this would check `operator[]` bounds so that you don't have to use the semantically incorrect `at()`). From what I've seen in the libstdc++ code the difference is mainly about precondition checking of ranges. For all iterator categories where `std::distance` is expensive, this is an expensive check, otherwise it's fairly cheap.

Reply:
Changed this to use `_GLIBCXX_ASSERTIONS` and `_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_EXTENSIVE` now.